app/routes/login.tsx

```diff
@@ -0,0 +1,87 @@
+import type { FC } from "react";
+import type { LoaderFunction, ActionFunction } from "@remix-run/node";
+import { GithubAuthProvider, signInWithPopup } from "firebase/auth";
+import { json, redirect } from "@remix-run/node";
+import { useLoaderData, useSubmit } from "@remix-run/react";
+import { commitSession, getSession } from "~/session.server";
+import RootLayout from "~/components/RootLayout";
+import Button from "~/components/Button";
+import { getFirebaseClientConfig } from "~/firebase.server";
+import { initFirebaseClient } from "~/firebase.client";
+
+export const loader: LoaderFunction = async ({ request }) => {
+  const session = await getSession(request.headers.get("Cookie"));
+
+  if (session.has("idToken")) {
+    // Redirect to the profile page if we're already signed in
+    return redirect("/profile");
+  }
+
+  const data = {
+    error: session.get("error"),
+    firebaseClientConfig: getFirebaseClientConfig(),
+  };
+
+  return json(data, {
+    headers: { "Set-Cookie": await commitSession(session) },
+  });
+};
+
+/**
+ * Handler triggered when a form is POSTed to this URL. Using the idToken field
+ * from the payload, we commit a new session on the server. This allows us to
+ * retrieve the user's session after the user closes the page or their browser.
+ */
+export const action: ActionFunction = async ({ request }) => {
```

Comment:
This is where we handle the form submission from our component. This code is executed on the server, parses the `idToken` from the payload, and stores it in a session cookie. If everything goes well, we redirect to the `/profile` page.